### ISSUE TITLE
Consider what happens when using `wallet_requestPermissions` to request snap install

### ISSUE THREAD

**Issue by FrederikBolding** [MetaMask/snaps-monorepo#980]
Consider what happens when using `wallet_requestPermissions` to request a snap to be installed.

This will most likely not work currently because snap install is a side effect.

### Outcome

1. Find out what happens when `wallet_requestPermissions` is used. If it is breaking, fix the issue. Otherwise, write unit tests for it.
2. Get data on what parts of the code are coupled, which parts of UI we *must* update to fix `wallet_requestPermissions`, this is important for #892

**Comment by FrederikBolding** [comment]
This issue will likely be more easily solved once we have completed #892. Since we want to unify to one popup, we could have that popup triggered by a permission request to `wallet_snap_foo` and trigger installation as a side effect.

**Comment by GuillaumeRx** [comment]
After looking at how we handle snap installation @FrederikBolding and I came up with those ideas:

### `wallet_requestPermissions` should have a side effect that calls `installSnaps()`.

In the extension - when an approval is closed - it waits to see if another approval request is triggered to prevent the window closing and displays immediately the second one. ( [ref.](https://github.com/MetaMask/metamask-extension/blob/13de51e748876707b06b144b0996b1e71738720b/ui/store/actions.js#L3130) ). We can utilize this to makes `installSnaps()` trigger a new approval 

### Approval updates
When `installSnaps()` is triggered we can immediatly trigger a new Approval showing a spinner indicating the snap is being fetched. We can't process the snap permissions before fetching it.

Although a method to update the approval with the snap data after we have successfully fetched is needed. This also means we can display a screen saying if a snap has succesfully installed or if it failed.

## TLDR

This means we need to add some functionality to the `ApprovalController`: 
- A method to update the approval in order to show the requested permissions as we don't have that data before the snap is fetched, this means we will need to add an `update()` method to the `ApprovalController` and feed the data to this method.
- A way to feed a side effect to an Approval that is called right after the approval is accepted.

**Comment by GuillaumeRx** [comment]
The research is done, the work is being split up in multiple tickets part of the #1119 epic.
